napari/components/layerlist.py

```diff
@@ -186,6 +188,60 @@ def toggle_selected_visibility(self):
             if layer.selected:
                 layer.visible = not layer.visible
 
+    @property
+    def _extent_world(self):
+        """(2, D) array: Extent of layers in world coordinates.
+
+        Default to 2D with (0, 512) min/ max values if no data is present.
+        """
+        if len(self) == 0:
+            min_v = [np.nan] * self.ndim
+            max_v = [np.nan] * self.ndim
+        else:
+            extrema = [l._extent_world for l in self]
+            mins = [e[0] for e in extrema]
+            maxs = [e[1] for e in extrema]
+
+            min_v = np.nanmin(
+                list(itertools.zip_longest(*mins, fillvalue=np.nan)), axis=1
+            )
+            max_v = np.nanmax(
+                list(itertools.zip_longest(*maxs, fillvalue=np.nan)), axis=1
+            )
+
+        min_vals = np.nan_to_num(min_v, nan=0)
+        max_vals = np.nan_to_num(max_v, nan=512)
+
+        return np.vstack([min_vals, max_vals])
+
+    @property
+    def _step_size(self):
+        """(D, ) array: Ideal step size between planes in world coordinates.
+
+        Computes the best step size that allows all data planes to be
+        sampled if moving through the full range of world coordinates.
+        The current implementation just takes the minimum scale, but a
+        better approach would use something like a greatest common divisor.
+        """
+        if len(self) == 0:
+            return np.ones(self.ndim)
+        else:
+            scales = [l.scale for l in self]
+            full_scales = np.array(
+                list(itertools.zip_longest(*scales, fillvalue=np.nan))
+            ).T
+            min_scales = np.nanmin(full_scales, axis=0)
+            return min_scales
+
+    @property
+    def ndim(self):
+        """int: Maximum dimensionality of layers.
+
+        Defaults to 2 if no data is present.
+        """
+        ndims = [l.ndim for l in self]
+        return max(ndims, default=2)
```

Comment:
```suggestion
        return max((l.ndim for l in self), default=2)
```
not that we need speed here, but just cause it's interesting: that's about 3x faster on my computer than first making a list.